lib/AFS/CellCC/Dump.pm

```diff
@@ -41,35 +41,42 @@ our @EXPORT_OK = qw(process_dumps);
 # currently just calculate a checksum for the dump blob, and update the
 # database to give the dump filename and metadata.
 sub
-_dump_success($$$$) {
-    my ($jobid, $dvref, $prev_state, $dump_fh) = @_;
+_dump_success($$$) {
+    my ($prev_state, $jobs_ref, $dump_fhs_ref) = @_;
+    my @jobs = @{$jobs_ref};
+    my @dump_fhs = @{$dump_fhs_ref};
     my $done_state = 'DUMP_DONE';
-    my $filesize = stat($dump_fh)->size;
+    my $filesize = stat($dump_fhs[0])->size;
 
-    my $base_file = basename($dump_fh->filename);
-
-    seek($dump_fh, 0, 0);
+    seek($dump_fhs[0], 0, 0);
 
+    foreach my $i (1 .. $#dump_fhs) {
+        File::Copy::copy($dump_fhs[0]->filename, $dump_fhs[$i]->filename);
+    }
     # Note that this checksum doesn't need to by cryptographically secure. md5
     # should be fine.
     my $algo = config_get('dump/checksum');
-    my $checksum = calc_checksum($dump_fh, $filesize, $algo, $jobid, $dvref, $prev_state);
-
-    update_job(jobid => $jobid,
-               dvref => $dvref,
-               from_state => $prev_state,
-               to_state => $done_state,
-               dump_fqdn => config_get('fqdn'),
-               dump_method => 'remctl',
-               dump_port => config_get('remctl/port'),
-               dump_filename => $base_file,
-               dump_checksum => $checksum,
-               dump_filesize => $filesize,
-               timeout => undef,
-               description => "Waiting to xfer dump file");
+    my $checksum = calc_checksum_jobs($dump_fhs[0], $filesize, $algo, $prev_state, @jobs);
+
+    foreach my $i (0 .. $#jobs) {
+        update_job(jobid => $jobs[$i]->{jobid},
```

Comment:
Ideally we'd have db primitives that can act on multiple jobs at once, so we don't need to issue separate queries like this. However, even without that this can be made a little more efficient just by wrapping this in a db_rw() call. See `startsync()` in `lib/AFS/CellCC.pm` for an example. That should let us issues these multiple calls in a single db connection and transaction, so it should be impossible to have only some of them fail.